src/main/java/org/oscm/app/shell/ShellController.java

```diff
@@ -123,47 +112,40 @@ void runVerificationScript(Configuration config) throws APPlatformException {
             return;
         }
 
-        Script script;
-        ScriptLogger scriptLogger = new ScriptLogger();
+        validator.validate(config, VERIFICATION_SCRIPT);
 
-        try {
-            script = new Script(verificationScript);
+        String instanceId = config.getSetting(INSTANCE_ID);
+        long timeout = Long.parseLong(config.getSetting(VERIFICATION_TIMEOUT));
+
+        try (Shell shell = new Shell()) {
+
+            Script script = new Script(verificationScript);
             script.loadContent();
             script.insertProvisioningSettings(config.getProvisioningSettings());
-        } catch (Exception e) {
-            throw new APPlatformException(e.getMessage());
-        }
 
-        try {
             ShellCommand command = new ShellCommand(script.getContent());
 
-            try (Shell shell = new Shell()) {
-                shell.lockShell(config.getSetting(INSTANCE_ID));
-                shell.runCommand(config.getSetting(INSTANCE_ID), command);
-                long ref = currentTimeMillis();
-                ShellStatus rc;
-                do {
-                    rc = shell.consumeOutput(config.getSetting(INSTANCE_ID));
-                    sleep(100);
-                } while (rc == RUNNING
-                        && currentTimeMillis() - ref < valueOf(config.getSetting(VERIFICATION_TIMEOUT)));
-
-                String shellOutput = shell.getOutput();
-                //JsonObject jsonOutput = shell.getResult();
-                //LOGGER.warn("Json output : " + jsonOutput);
-                scriptLogger.logOutputFromScript(config, "VERIFICATION_SCRIPT", shellOutput);
-
-                Pattern p = compile(format(".*%s=(.*?)$", VERIFICATION_MESSAGE), MULTILINE);
-                Matcher matcher = p.matcher(shellOutput);
-
-                if (matcher.find()) {
-                    throw new APPlatformException("Verification failed: " + matcher.group(1));
-                }
+            shell.lockShell(instanceId);
+            shell.runCommand(instanceId, command);
+
+            ShellStatus status;
+            long startTime = currentTimeMillis();
+            do {
+                status = shell.consumeOutput(instanceId);
+                Thread.sleep(1000);
+            } while (status == RUNNING && currentTimeMillis() - startTime < timeout);
+
+            ShellResult result = shell.getResult();
+            String message = result.getMessage();
+
```

Comment:
Q: What happens on timeout? Just stops polling the status?

Reply:
It will end up with error ( Invalid JSON returned after script execution) as the resulted JSON will not be available and therefore invalid